contents/performance/PerformanceProfiling.md

```diff
@@ -0,0 +1,137 @@
+# Performance Profiling
+
+Author: [Ong Heng Le](https://github.com/initialshl)
+
+## Overview
+
+A performance profiler is a tool which collects data (such as function timings) about 
+your program to identify the areas with performance issues, commonly used in code 
+optimization. There are two common types of performance profiling methods.
+
+* **CPU sampling**: Collects *samples* at fixed intervals, which provides an overview of your program's performance
+* **Instrumentation**: Collects detailed *elapsed times* for your program's function calls
+
+Learn [which profiling method to use for your program](https://blogs.msdn.microsoft.com/ejarvi/2005/04/07/the-choice-between-sampling-and-instrumentation/) 
+when choosing between CPU sampling and instrumentation.
+
+### Resources used:
+
+1. What is a software profiler?: [Profiling Overview](https://msdn.microsoft.com/en-us/library/bb384493(v=vs.110).aspx)
+1. Common performance profiling methods: [Understanding Performance Collection Methods](https://msdn.microsoft.com/en-us/library/dd264994.aspx)
+
+## Performance Profiling For The First Time
+
+This article provides resources to help you identify the performance issues as you 
+perform profiling on your program for the first time. The general steps in profiling a 
+program for the first time are:
+
+1. Run a performance profiling session 
+1. View the performance report 
+1. Trace down to the problem 
+1. Identify areas for improvements upwards 
+
+Most profilers are similar in their functionalities, user interface, and use of 
+technical terms. You may adapt the tutorials in this article to your preferred profiling 
+tools on your own project. The following tutorials are available in this article: 
+
+* [Profiling a Desktop Application In Visual Studio 2015](#profiling-a-desktop-application-in-visual-studio-2015)
+
+### Recommended resources:
+
+1. Learn the best practices in profiling: [Advanced Profiling: Theory in Practice with NetBeans IDE](https://netbeans.org/community/magazine/html/04/profiler.html)
+
+# Profiling a Desktop Application In Visual Studio 2015
+
+> This tutorial demonstrates using the profiler included in [Visual Studio 2015](https://www.visualstudio.com/downloads/). 
```

Comment:
This should no longer be in the infobox because this section is now talking about VS2015.